File: frontend/services/enrollmentService.js
import api from "./api";

export const getEnrollments = (params) => {
  return api.get("/api/enrollments", {
    params: {
      page: params.page,
      size: params.size,
      search: params.search,
      sortBy: params.sortBy,
      sortDir: params.sortDir,
    },
  });
};

export const createEnrollment = (data) => {
  return api.post("/api/enrollments", data);
};

export const updateEnrollment = (id, data) => {
  return api.put(`/api/enrollments/${id}`, data);
};

export const deleteEnrollment = (id) => {
  return api.delete(`/api/enrollments/${id}`);
};

File: Backend/src/main/java/com/assignment/courseManagementSystem/services/contract/EnrollmentService.java
package com.assignment.courseManagementSystem.services.contract;

import com.assignment.courseManagementSystem.dto.EnrollmentDto;
import com.assignment.courseManagementSystem.entities.Course;
import org.springframework.data.domain.Page;

import java.util.List;

public interface EnrollmentService {

    EnrollmentDto createEnrollment(EnrollmentDto dto);


    EnrollmentDto getEnrollmentById(Long id);

    Page<EnrollmentDto> getEnrollments(
            String search,
            int page,
            int size,
            String sortBy,
            String sortDir
    );

    EnrollmentDto updateEnrollment(Long id, EnrollmentDto dto);

    void deleteEnrollment(Long id);
}


